app/src/main/java/com/jalasoft/routesapp/ui/auth/registerUser/view/RegisterUserFragment.kt

```diff
@@ -0,0 +1,122 @@
+package com.jalasoft.routesapp.ui.auth.registerUser.view
+
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.view.isVisible
+import androidx.fragment.app.viewModels
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.firebase.FirebaseApp
+import com.jalasoft.routesapp.R
+import com.jalasoft.routesapp.databinding.FragmentRegisterUserBinding
+import com.jalasoft.routesapp.ui.auth.registerUser.viewModel.RegisterUserViewModel
+
+class RegisterUserFragment : Fragment() {
+
+    private var _binding: FragmentRegisterUserBinding? = null
+    private val binding get() = _binding!!
+    private val viewModel: RegisterUserViewModel by viewModels()
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        FirebaseApp.initializeApp(context)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentRegisterUserBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.btnRegRegister.setOnClickListener {
+            addUser()
+        }
+    }
+
+    private fun addUser() {
+        if (validateFields()) {
+            showProgress(true)
+            val name = binding.etRegName.text.toString()
+            val email = binding.etRegEmail.text.toString()
+            val password = binding.etRegPassword.text.toString()
+
+            viewModel.registerUserAuth(name, email, password, { _ ->
+                binding.txtRegError.isVisible = false
```

Comment:
This is wrong, success and failure should be handled on the viewmodel and update variables, here you just need to listen to variable changes

Reply:
Done!

Reply:
This is still wrong, this logic should be handled on the viewmodel, and you should be observing variable changes on the Fragment. I don't see a place where this is happening, you´re trying to imitate that with the callback but it isn't that way.

![imagen](https://user-images.githubusercontent.com/1596425/183504207-cb16ce32-f4ee-4fca-91b0-7f61afc3199b.png)

Example from this link:
https://developer.android.com/topic/libraries/architecture/livedata#observe_livedata_objects

Reply:
i think now it is good i add the observer for the viewModel you can check it